File: packages/face-detection/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/facedetection/classes/ProcessImageOptions.java
package io.capawesome.capacitorjs.plugins.mlkit.facedetection.classes;

import com.google.mlkit.vision.common.InputImage;
import com.google.mlkit.vision.face.FaceDetectorOptions;

public class ProcessImageOptions {

    private InputImage inputImage;

    @FaceDetectorOptions.PerformanceMode
    private int performanceMode;

    @FaceDetectorOptions.LandmarkMode
    private int landmarkMode;

    @FaceDetectorOptions.ContourMode
    private int contourMode;

    @FaceDetectorOptions.ClassificationMode
    private int classificationMode;

    private float minFaceSize;
    private boolean enableTracking;

    public ProcessImageOptions(
        InputImage inputImage,
        @FaceDetectorOptions.PerformanceMode int performanceMode,
        @FaceDetectorOptions.LandmarkMode int landmarkMode,
        @FaceDetectorOptions.ContourMode int contourMode,
        @FaceDetectorOptions.ClassificationMode int classificationMode,
        float minFaceSize,
        boolean enableTracking
    ) {
        this.inputImage = inputImage;
        this.performanceMode = performanceMode;
        this.landmarkMode = landmarkMode;
        this.contourMode = contourMode;
        this.classificationMode = classificationMode;
        this.minFaceSize = minFaceSize;
        this.enableTracking = enableTracking;
    }

    public InputImage getInputImage() {
        return inputImage;
    }

    public int getPerformanceMode() {
        return performanceMode;
    }

    public int getLandmarkMode() {
        return landmarkMode;
    }

    public int getContourMode() {
        return contourMode;
    }

    public int getClassificationMode() {
        return classificationMode;
    }

    public float getMinFaceSize() {
        return minFaceSize;
    }

    public boolean isTrackingEnabled() {
        return enableTracking;
    }
}

File: packages/face-detection/ios/Plugin/Classes/ProcessImageOptions.swift
import Foundation
import MLKitVision

@objc class ProcessImageOptions: NSObject {
    private var visionImage: VisionImage
    private var performanceMode: Int
    private var landmarkMode: Int
    private var contourMode: Int
    private var classificationMode: Int
    private var minFaceSize: Float
    private var enableTracking: Bool

    init(
        visionImage: VisionImage,
        performanceMode: Int,
        landmarkMode: Int,
        contourMode: Int,
        classificationMode: Int,
        minFaceSize: Float,
        enableTracking: Bool
    ) {
        self.visionImage = visionImage
        self.performanceMode = performanceMode
        self.landmarkMode = landmarkMode
        self.contourMode = contourMode
        self.classificationMode = classificationMode
        self.minFaceSize = minFaceSize
        self.enableTracking = enableTracking
    }

    func getVisionImage() -> VisionImage {
        return visionImage
    }

    func getPerformanceMode() -> Int {
        return performanceMode
    }

    func getLandmarkMode() -> Int {
        return landmarkMode
    }

    func getContourMode() -> Int {
        return contourMode
    }

    func getClassificationMode() -> Int {
        return classificationMode
    }

    func getMinFaceSize() -> Float {
        return minFaceSize
    }

    func isTrackingEnabled() -> Bool {
        return enableTracking
    }
}
